packages/shared-core/src/filters.ts

```diff
@@ -106,12 +110,35 @@ const removeKeyNumbering = key => {
   }
 }
 
+type Filter = {
+  operator: keyof Query
+  field: string
+  type: any
+  value: any
+  externalType: keyof typeof SqlNumberTypeRangeMap
```

Comment:
NAB: What does `keyof typeof SqlNumberTypeRangeMap` do?

Reply:
It says the value should be a key of the selected type. Given the type
<img width="282" alt="image" src="https://user-images.githubusercontent.com/15987277/223666219-d96dc4c3-94e2-4cdc-8f82-325e90ccb760.png">
the `externalType` must be the following:
<img width="282" alt="image" src="https://user-images.githubusercontent.com/15987277/223666332-6efef032-354c-451e-bd9b-336daa4c4071.png">

Reply:
Ah yes I see - you only need `keyof Query` because Query is a **type** already. Thanks!